criu/kerndat.c

```diff
@@ -226,6 +235,46 @@ static int kerndat_get_shmemdev(void)
 	return -1;
 }
 
+static int kerndat_get_hugetlb_dev(void)
+{
+	void *map;
+	int i, flag;
+	unsigned long long size;
+	dev_t dev;
+
+	for (i = 0; i < HUGETLB_MAX; i++) {
+		/* This hugetlb size is not supported or the dev's number has been collected */
+		if (kdat.hugetlb_dev[i])
+			continue;
+
+		size = hugetlb_info[i].size;
+		flag = hugetlb_info[i].flag;
+		map = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | flag, 0, 0);
+		if (map == MAP_FAILED && errno == EINVAL) {
+			kdat.hugetlb_dev[i] = (dev_t)-1;
+			continue;
+		}
+
+		if (map == MAP_FAILED && errno == ENOMEM) {
+			pr_info("Hugetlb size %llu Mb is supported but cannot get dev's number\n", size >> 20);
+			continue;
+		} else if (map == MAP_FAILED) {
```

Comment:
`if (map == MAP_FAILED)` repeats three times here. Consider nesting `errno` checks under it.